django/db/backends/mysql/operations.py

```diff
@@ -347,8 +347,12 @@ def explain_query_prefix(self, format=None, **options):
         if analyze and self.connection.features.supports_explain_analyze:
             # MariaDB uses ANALYZE instead of EXPLAIN ANALYZE.
             prefix = 'ANALYZE' if self.connection.mysql_is_mariadb else prefix + ' ANALYZE'
-        if format and not (analyze and not self.connection.mysql_is_mariadb):
-            # Only MariaDB supports the analyze option with formats.
+        if format and not (
+            analyze and
+            not self.connection.mysql_is_mariadb and
+            self.connection.mysql_version < (8, 0, 21)
+        ):
+            # MariaDB and MySQL 8.0.21+ support the analyze option with format.
```

Comment:
MySQL 8.0.21+ still supports only the `TREE` format which is used by default, so I don't see much value in this change:

_"In MySQL 8.0.21 and later, this can optionally be specified explicitly using FORMAT=TREE; **formats other than TREE remain unsupported**."_

this can also be misleading to users.

Reply:
That is true. I guess syntactically they've lifted the limitation. But as we validate the value of `FORMAT`, we'd then have two sets of supported values based on whether `ANALYZE` was used.

Let's pan this for later. It looks as though support for `FORMAT=JSON` was proposed for the future in [WL#4168](https://dev.mysql.com/worklog/task/?id=4168) if you look under the "High Level Architecture" tab, so this may come in a future release. (I found that link from bug report [#97282](https://bugs.mysql.com/bug.php?id=97282).)

Even if they support `JSON` but not `TEXT`/`TRADITIONAL`, then I think it will be worth revisiting.